scipy/special/_basic.py

```diff
@@ -2953,6 +2956,137 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def stirling2(N, K, exact=True):
+    """Generate Stirling number(s) of the second kind.
+
+        Stirling numbers of the second kind count the number of ways to
+        partition a set with N elements into K non-empty subsets.
+
+        The values this function returns are calculated using a dynamic
+        program which avoids redundant computation across the subproblems
+        in the solution.
```

Comment:
Good to make it even clearer the extent to which redundant computation is avoided.

```suggestion
        program which avoids redundant computation across the subproblems
        in the solution. For array-like input, this implementation also 
        avoids redundant computation across the different Stirling number
        calculations.
```